code/game/objects/structures/displaycase.dm

```diff
@@ -184,7 +185,12 @@ var/global/list/captain_display_cases = list()
 			playsound(get_turf(src), "shatter", 70, 1)
 			update_icon()
 			spawn(0)
-				burglar_alarm()
+				if (!alarm_needs_power)

```

Comment:
Honestly i could be wrong but this feels like the check would be better handled on the proc itself to see if the area is powered or not.